src/wp-admin/includes/update-core.php

```diff
@@ -7,6 +7,15 @@
  * @since 2.7.0
  */
 
+/**
+ * Preload files from the current version into memory that may be
+ * needed during the update process, and may not be available
+ * after files from the updated version have been installed.
+ *
+ * @since 5.9.0
+ */
+require_once ABSPATH . 'wp-includes/Requests/Exception.php';
```

Comment:
Requests library was introduced in WordPress 4.6. Upgrading from < 4.6 would result in a fatal error.

```suggestion
global $wp_version;
if ( version_compare( $wp_version, '4.6', '>=' ) ) {
	require_once ABSPATH . 'wp-includes/Requests/Exception.php';
}
```
Adding a version check here provides a guard and grouping for any other files.

Reply:
Q: Is `$wp_version` still set to the current ("from") version at this point?

A: Loads the current ("from") version `wp-includes/version.php` file at the start of `Core_Upgrader::upgrade()`.

Reply:
good catch on the earliest version Requests lib was introduced in